internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -426,6 +426,49 @@ func (r *BMACReconciler) reconcileAgentInventory(bmh *bmh_v1alpha1.BareMetalHost
 
 }
 
+// Utility to verify whether a BMH should be reconciled based on the InfraEnv
+//
+// This function verifies three things:
+//
+// 1. InfraEnv exists
+// 2. InfraEnv's ISODownloadURL exists
+// 3. The BMH.Spec.URL is the same to the InfraEnv's ISODownload URL
+//
+// If the three checks above are true, then the reconcile won't happen
+// and the reconcileBMH function will return.
+//
+// Note that this function is also used to decide whether a BMH reconcile should be queued
+// or not. This will help queuing fewer reconciles when we know the BMH is not ready.
+//
+// The function returns 2 booleans:
+//
+// 1. Should we proceed with the BMH's reconcile
+// 2. Should the full `Reconcile` stop as well
+func shouldReconcileBMH(bmh *bmh_v1alpha1.BareMetalHost, infraEnv *aiv1beta1.InfraEnv) (bool, bool) {
+	// The global `Reconcile` flow is not stopped in this case
+	// as we may be reconciling a BMH in the spokeCluster that
+	// doesn't have an InfraEnv.
```

Comment:
Got a bit confused for a second so I would appreciate someone verifying this logic:

/cc @tsorya @mhrivnak

Reply:
not sure what it means, we are not reconciling spoke cluster no? we just creating it as part of reconciliation process of hub bmh when we need to install day2?

Reply:
> not sure what it means, we are not reconciling spoke cluster no? we just creating it as part of reconciliation process of hub bmh when we need to install day2?

This is where I am coming from.

I think this check could be changed to return stop the `Reconcile` loop in the hub . If there's no InfraEnv then there is no point in Reconciling a BMH at all